hokusai/staging.yml

```diff
@@ -6,8 +6,8 @@ metadata:
 spec:
   strategy:
     rollingUpdate:
-      maxSurge: 0
-      maxUnavailable: 20%
```

Comment:
I understand `maxUnavailable` as inversely, at least 80% pods will be available during rollout as [per these docs](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#max-unavailable)

Is the issue more so the horizontal pod autoscaler's setting `minReplicas: 1` conflicting with deployment totoal replicas during rollout?

Reply:
I think the problem is when actual pod count is at min which is 1. In that situation, because k8s is not allowed to surge (maxSurge 0), during deployment, k8s must delete the one and only pod first, before it can add a new pod. I am not 100% sure that it is happening but it seems very possible based on all the settings.